src/game.cpp

```diff
@@ -6502,12 +6502,16 @@ void game::monmove()
 {
     cleanup_dead();
 
-    // monster::plan() needs to know about all monsters with nonzero friendliness.
-    // We'll build this list once (instead of once per monster) for speed.
-    std::vector<int> friendlies;
+    // monster::plan() needs to know about all monsters on the same team as the monster
+    static const std::string playerfaction = "PLAYER";
+    std::map< std::string, std::set< int > > monster_factions;
```

Comment:
You can try to store a fraction pointer here instead of the fraction string and later only compare the pointer, that might speedup the process.